gptel.el

```diff
@@ -1116,6 +1122,24 @@ Valid JSON unless NO-JSON is t."
 
 ;;; Saving and restoring state
 
+(defun gptel--restore-props (bounds-vals)
+  "Restore text properties from BOUNDS-VALS.
+BOUND-VALS is a list of (BEG END . VALUE)"
+  (mapc (lambda (bound-val)
+          (pcase bound-val
+            (`(,beg ,end . ,value)
+             (add-text-properties
+              beg end
+              (pcase value
+                ('response '(gptel response front-sticky (gptel)))
+                (`('tool-call . ,_) `(gptel ,value))
+                (`('tool-result . ,_) `(gptel ,value))
+                ('ignore '(gptel rear-nonsticky '(gptel))))))
```

Comment:
The `gptel` text-property is globally `rear-nonsticky` (see `text-property-default-nonsticky`) so this is not required.